component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+        }
+
+        @Override
+        public void onFileCreate(final File file) {
+            log.debug(file.getAbsolutePath() + " was created.");
+            initialMap.put(file.getAbsolutePath(), file.length());
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(file, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onFileChange(final File file) {
+            log.debug(file.getAbsolutePath() + " was modified.");
+            synchronized (this) {
+                if (!recentFileSizeMap.containsKey(file.getAbsolutePath())) {
+                    recentFileSizeMap.put(file.getAbsolutePath(), file.length());
+                    sourceEventListener.onEvent
+                            (getFileHandlerEvent(file, listeningFileUri, STATUS_PROCESS), null);
+                    ThreadOnContinuousChange threadOnContinuousChange = new ThreadOnContinuousChange(file);
+                    siddhiAppContext.getScheduledExecutorService().schedule(threadOnContinuousChange,
+                            monitoringInterval, TimeUnit.MILLISECONDS);
+                    threadList.add(threadOnContinuousChange);
+                }
+            }
+        }
+
+        @Override
+        public void onFileDelete(final File file) {
+            log.debug(file.getAbsolutePath() + " was deleted.");
+            initialMap.remove(file.getAbsolutePath());
+            sourceEventListener.onEvent
+                    (getFileHandlerEvent(file, listeningFileUri, STATUS_DONE), null);
+        }
+
+        @Override
+        public void onStop(final FileAlterationObserver observer) {
+
+        }
+    }
+
+    public void initiateFileAlterationObserver() {
+        FileAlterationObserver observer = new FileAlterationObserver(listeningDirUri);
+        observer.addListener(new FileAlterationImpl());
+        monitor = new FileAlterationMonitor(monitoringInterval);
+        monitor.addObserver(observer);
+        try {
+            monitor.start();
+            log.debug("Directory monitoring has been started for folder : " + listeningUri + " .");
+        } catch (Exception e) {
+            throw new SiddhiAppRuntimeException("Exception occurred when starting server to monitor "
+                    + listeningUri + ".", e);
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        if (monitor != null) {
+            try {
+                monitor.stop();
+                initialMap.clear();
+                log.debug("Directory monitoring has been stopped for folder : " + listeningUri + " .");
+            } catch (Exception e) {
+                log.error("Exception occurred when stopping server  while monitoring " + listeningUri + " .", e);
+            }
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public void pause() {
+        if (monitor != null) {
+            threadList.forEach(ThreadOnContinuousChange::pause);
```

Comment:
rename this thread to a meaningful name

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/3f9a2a0eb50655821dd322688aa034be33ef8860